runtime/vm/jnifield.cpp

```diff
@@ -414,11 +413,10 @@ setObjectField(JNIEnv *env, jobject obj, jfieldID fieldID, jobject valueRef)
 
 	VM_VMAccess::inlineEnterVMFromJNI(vmThread);
 
-	object = J9_JNI_UNWRAP_REFERENCE(obj);
-
 	/* A NULL value is ok */
+	FieldEvents::triggerSetEvents(vmThread, j9FieldID, obj, (NULL == valueRef)? NULL : (void*)valueRef);
```

Comment:
If we pass NULL when valueRef is NULL, why bother checking?

Reply:
Currently, the hook requires a non-null valueAddress, which this breaks, so this is actually incorrect.